crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -455,6 +456,32 @@ def services
                   node["pacemaker"]["founder"] == node[:fqdn])
             nova_node = node
           end
+          if node.roles.include?("database-server") &&
+              (!node.roles.include?("pacemaker-cluster-member") ||
+                  node["pacemaker"]["founder"] == node[:fqdn])
+            database_node = node
+          end
+        end
+
+        begin
+          unless database_node.nil?
+            database_node.wait_for_script_to_finish(
```

Comment:
I think this should be called before this "stopping" loop. If monasca is running on a node which is before the database-server in `upgrade_nodes` list then the monasca database will already be stopped when we try to dump it from database node.

Reply:
Not a problem anymore: I moved back to dumping from the monasca-server node (since that's where we'll restore from later - the restore happens in a recipe running there as well, so it makes sense to put have the dump available on that machine).